float32 theta_d
float32 phi_d